### PR TITLE
builtin: add degrees built-in function

### PR DESCRIPTION
**Pull request by arthuryangcs**
builtin: add degrees built-in function

### CODE REVIEW

File: expression/builtin_math_test.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ import (
 	"github.com/pingcap/tidb/util/testleak"
 	"github.com/pingcap/tidb/util/testutil"
 	"github.com/pingcap/tidb/util/types"
+	"math"
```
Comment (shenli) [inline expression/builtin_math_test.go]: Put this before "gitub.com/xxx" and leave a blank line between them.

File: expression/builtin_math.go [hunk 2]
```diff
@@ -793,7 +793,25 @@ type builtinDegreesSig struct {
...
-	return d, errFunctionNotExists.GenByArgs("degrees")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return types.Datum{}, errors.Trace(err)
+	}
+
+	sc := b.ctx.GetSessionVars().StmtCtx
+
+	if args[0].IsNull() {
+		return args[0], nil
```
Comment (shenli) [inline expression/builtin_math.go]: return d, nil

File: expression/builtin_math_test.go [hunk 3]
```diff
@@ -351,3 +352,29 @@ func (s *testEvaluatorSuite) TestSqrt(c *C) {
...
+	Dtbl := tblToDtbl(tbl)
+
+	for _, t := range Dtbl {
+		fc := funcs[ast.Degrees]
+		f, err := fc.getFunction(datumsToConstants(t["Arg"]), s.ctx)
+		c.Assert(err, IsNil)
+		v, err := f.eval(nil)
+		c.Assert(err, IsNil)
+		c.Assert(v, DeepEquals, t["Ret"][0], Commentf("arg:%v", t["arg"]))
+	}
```
Comment (shenli) [inline expression/builtin_math_test.go]: Please also add some invalid input tests. Such as degree("abc").

**Comment by shenli** [comment]
@arthuryangcs Thanks for your PR! Please address the comments and resolve the conflict.

File: expression/builtin_math_test.go (hunk 3 above)
Comment (arthuryangcs) [inline expression/builtin_math_test.go]: Add some input tests.
Is it enough?

Comment (arthuryangcs) [inline expression/builtin_math_test.go]: I found that there are some same problems in sqrt built-in function

**Comment by arthuryangcs** [comment]
Already addressed the comments and resolved the conflict.

File: expression/builtin_math.go [hunk 4]
```diff
@@ -852,7 +852,28 @@ type builtinDegreesSig struct {
...
+	if args[0].IsNull() {
+		d.SetNull()
+		return d, nil
+	}
+
+	tmpIT := sc.IgnoreTruncate
+	sc.IgnoreTruncate = true
+	x, err := args[0].ToFloat64(sc)
+	if err != nil {
+		sc.IgnoreTruncate = tmpIT
```
Comment (shenli) [inline expression/builtin_math.go]: Move this before the if statement, then you can save line 874.

Comment (arthuryangcs) [inline expression/builtin_math.go]: I made a very very wrong choice that I used sqrt built-in function as a reference.

Comment (shenli) [inline expression/builtin_math.go]: :) Feel free to refine `sqrt` function code.

**Comment by shenli** [comment]
LGTM
@arthuryangcs Please resolve the conflicts.

**Comment by arthuryangcs** [comment]
@shenli Already solved the conflicts again.

**Comment by XuHuaiyu** [comment]
@arthuryangcs 
If possible, please check and register the builtin function you decide to implement in [issue](https://github.com/pingcap/tidb/issues/2863) to let others know, thank you!

File: expression/builtin_math.go [hunk 5]
```diff
@@ -853,7 +853,27 @@ type builtinDegreesSig struct {
...
+
+	sc := b.ctx.GetSessionVars().StmtCtx
+
+	if args[0].IsNull() {
+		d.SetNull()
+		return d, nil
+	}
+
+	tmpIT := sc.IgnoreTruncate
+	sc.IgnoreTruncate = true
```
Comment (coocood) [inline expression/builtin_math.go]: Please do not modify the `IgnoreTruncate` in implementation, modify it in test code.

Comment (arthuryangcs) [inline expression/builtin_math.go]: @coocood 
I don't know how to modify it.
In built-in function `Floor`, it modify the `IgnoreTruncate` in implementation, too.
Can you make an example?

Comment (coocood) [inline expression/builtin_math.go]: Would you please fix that too?
In test code, the `sc` can be obtained by

```sc := s.ctx.GetSessionVars().StmtCtx```

You can modify it and run the test case, then set it back to the original value.

Comment (arthuryangcs) [inline expression/builtin_math.go]: @coocood 
Already modified it, is it right?

**Comment by coocood** [comment]
LGTM
Thank you!

**Pull request merged**